builtin/providers/cloudstack/resource_cloudstack_instance.go

```diff
@@ -186,6 +195,7 @@ func resourceCloudStackInstanceRead(d *schema.ResourceData, meta interface{}) er
 	d.Set("display_name", vm.Displayname)
 	d.Set("ipaddress", vm.Nic[0].Ipaddress)
 	d.Set("zone", vm.Zonename)
+	//NB cloudstack sometimes sends back the wrong keypair name, so dont update it
```

Comment:
So we chased it and indeed it seems to be buggy and/or questionable at the very least.

Still I think it's such a corner case, that I would prefer to read back the associated value. At the same time we will work on fixing this (or making it more robust) in CS. For that it would be great if you can log an issue: http://issues.apache.org/jira

So what do you think about reading it back? Should be safe for normal operations right?